Zend/zend_fibers.c

```diff
@@ -141,7 +141,7 @@ typedef struct {
 
 /* These functions are defined in assembler files provided by boost.context (located in "Zend/asm"). */
 extern void *make_fcontext(void *sp, size_t size, void (*fn)(boost_context_data));
-extern boost_context_data jump_fcontext(void *to, zend_fiber_transfer *transfer);
+extern boost_context_data jump_fcontext(void *to, zend_fiber_transfer *transfer) __attribute__ ((__indirect_return__));
```

Comment:
`__attribute__ ((__indirect_return__))` is not necessarily supported (see the failing CI builds), and as such must not be applied uncoditionally.

Reply:
Maybe it is okay to only enable `shstk` if `__attribute__((indirect_return))` (or equivalent) is supported by the compiler. Those interested in best security practices would need to use a newish compiler version, but that may be okay.

I wouldn't know if something like `__attribute__((indirect_return))` is supported by MSVC. [CET support](https://docs.microsoft.com/en-us/cpp/build/reference/cetcompat?view=msvc-170) is generally available there.

Reply:
Based on my current understanding (maybe it's not correct), there are no such attribute for shadow stack.
Basically, the compiler such as gcc take care of shadow stack. No special treatment is needed for application.

But Fiber or applications use [makecontext()](https://man7.org/linux/man-pages/man3/makecontext.3.html) are exception. They make context (i.e., they maintain stack) by themselves instead of gcc/libc. So they also need creating a shadow stack which requires kernel support. Such kernel patch is under review..